File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    unordered_map<string, int> u;
    for(string c : completion) {
        u[c]++;
    }
    
    for(string p : participant) {
        if(u[p] == 0) {
            return p;
        }
        u[p]--;
    }
    return "";
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> s = new HashMap<>();
        for(String c : completion) {
            s.put(c, s.getOrDefault(c, 0) + 1);
        }
        
        for(String p : participant) {
            if (!s.containsKey(p) ||  s.get(p) == 0) {
                return p;
            }
            s.put(p, s.get(p) - 1);
        }
        return "";
    }
}